.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_size/aws_cryptosdk_enc_ctx_size_harness.c

```diff
@@ -31,22 +31,29 @@ void hash_iterator_generator(struct aws_hash_iter *new_iter, const struct aws_ha
 void hash_iterator_generator2(struct aws_hash_iter *new_iter, const struct aws_hash_iter *old_iter) {
     (void)old_iter;
     if (new_iter->status == AWS_HASH_ITER_STATUS_READY_FOR_USE) {
-        new_iter->element.key   = malloc(sizeof(struct aws_string));
+        new_iter->element.key = malloc(sizeof(struct aws_string));
+        __CPROVER_assume(new_iter->element.key != NULL);
         new_iter->element.value = malloc(sizeof(struct aws_string));
+        __CPROVER_assume(new_iter->element.value != NULL);
```

Comment:
Why not assume an invariant here (i.e., `aws_string_is_valid()`)? What is the corresponding precondition in the function?

Reply:
Same as above.